File: Archive 4/CoreDataExample/CoreDataExample/ViewController.m
//
//  ViewController.m
//  CoreDataExample
//
//  Created by Green on 7/9/15.
//  Copyright (c) 2015 Greens. All rights reserved.
//

#import "ViewController.h"
#import "AppDelegate.h"
#import "Student.h"
#import "SecondViewController.h"
@interface ViewController ()
{
    AppDelegate *appDelegate;
    NSFetchRequest *fetchRequest;
    NSArray  *array;
    Student  *stud ;
    NSArray *array_two;
}
@end

@implementation ViewController
@synthesize mangedobjectContext    ;
- (void)viewDidLoad {
    
    appDelegate =   (AppDelegate *) [[UIApplication sharedApplication]delegate];
    mangedobjectContext   = appDelegate.managedObjectContext;
    
    
    
   
    fetchRequest = [[NSFetchRequest alloc] initWithEntityName:@"Student"];
    array = [mangedobjectContext executeFetchRequest:fetchRequest error:nil];

    
    
    
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)saveAction:(id)sender {
    Student  *stud  =   [NSEntityDescription insertNewObjectForEntityForName:@"Student" inManagedObjectContext:mangedobjectContext];
    stud.name  = name.text;
    NSNumberFormatter * myNumFormatter = [[NSNumberFormatter alloc] init];
 NSNumber *tempNum = [myNumFormatter numberFromString:age.text];
    
    stud.age  = tempNum;
    stud.lang=lang.text;
    
    NSError *error;
    
    if([mangedobjectContext save:&error])
    {
        NSLog(@"Saved Suucesssfully");
    }
    else
    {
        NSLog(@"Saved  unsuccess");
    }
    
    

    
}


- (IBAction)showAction:(id)sender {
    fetchRequest = [[NSFetchRequest alloc] initWithEntityName:@"Student"];
    array = [mangedobjectContext executeFetchRequest:fetchRequest error:nil];
    NSLog(@"%@",array);
    NSLog(@"%@",[[array valueForKey:@"name"]objectAtIndex:1]);
    NSLog(@"%@",[array valueForKey:@"age"]);
    NSLog(@"%@",[array valueForKey:@"name"]);
    NSLog(@"%@",[array valueForKey:@"lang"]);
}

- (IBAction)predicateAction:(id)sender {
    NSString *langID=@"English";
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"lang==%@",
                              langID];
    [fetchRequest setPredicate:predicate];
    NSArray *array_one= [mangedobjectContext executeFetchRequest:fetchRequest error:nil];
    

    NSLog(@"Fetch  age Array 1%@",[array_one valueForKey:@"age"]);
    NSLog(@"Fetch name Array 1%@",[array_one valueForKey:@"name"]);
     NSLog(@"Fetch lang Array 1%@",[array_one valueForKey:@"lang"]);
        NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"name" ascending:YES];
        [fetchRequest setSortDescriptors:[NSArray arrayWithObject:sortDescriptor]];
     array_two= [mangedobjectContext executeFetchRequest:fetchRequest error:nil];
    NSLog(@"Fetch  age%@",[array_two valueForKey:@"age"]);
    NSLog(@"Fetch name%@",[array_two valueForKey:@"name"]);
    NSLog(@"Fetch lang%@",[array_two valueForKey:@"lang"]);
    
}
- (IBAction)nextAction:(id)sender{
    SecondViewController *third=[self.storyboard instantiateViewControllerWithIdentifier:@"second"];
    third.modalTransitionStyle =UIModalTransitionStylePartialCurl;
    third.array_secondView=[array_two copy];
    [self presentModalViewController:third animated:YES];
}
@end

File: Archive 4/swift/Demo/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by Green on 4/12/15.
//  Copyright (c) 2015 Green. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        
        var  a = 5
        var  b = 6
        
        var c = 0
        c = a+b
        
        
        println("Value a is \(a)")
        println("Value b is \(b)")
        println("The Value is \(c)")
        
        var array = ["one","vvvv","Three","Four","5","6"]
        
        println(array[3])
        
        
        var dict = ["1":"One","2":"Two","3":"Three"]
        
         println(dict["4"])
        
//        //3.Iterate through dictionary
//        for(key,value) in dict1
//        {
//            println("key: \(key) value:\(value)")
//        }
//      
        
        var  x = 0;
        
         var  first  = "5"
        
        for var i = 0 ; i<5; i++
        {
            
            if(first == array[i])
            {
                x++;
                
            }
            else
            {
                
            }
        }
        
//        
//        var second = "Swift"
//        
//        
//        if(first  == second)
//        {
//            println("equal")
//        }
//        else
//        {
//            println("non equal")
//
//        }
//
//        let oneString = "Hello"
//        let anotherString = "Hello"
//        
//        if (oneString == anotherString) {
//            println("Strings are equal")
//        }
//            else {
//                println("Strings are not equal")
//        }
        
//        self.fn(5.1 , 66)
//
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    
    //func AddAndSubtractTenFromNumber(number: Int) -> (small: Int, large: Int)
    
//    func fn(a: Float, b: int)->(nt)
//    {
//        
//        println(a)
//        return a
//    }
//    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

